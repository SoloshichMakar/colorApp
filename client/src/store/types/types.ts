export interface ITask {
  taskColor: string;
  taskText: string;
  id: string;
  completed: boolean;
}

export interface IState {
  tasks: Array<ITask>;
  inputText: string; //TODO: inputText
  chosenColor: string;
}

export interface IAction {
  type: string;
  value?: string;
  newTask?: ITask;
  inputText?: string; //TODO: inputText
  deleteId?: string;
  completed?: boolean;
  completedTaskId?: string;
  taskData?:Array<ITask>
  userId?: number
}


export interface ILogin {
  email: string,
  password: string,
  isAuthenticated: boolean,
  userId: number,
  message: string
}

export interface IStateRegistration {
  email: string,
  password: string,
  confirmPassword: string,
  message: string
}

export interface IActionLogin {
  type: string;
  email?: string,
  password?: string,
  checkedUser?:number,
  errorMessage?: string
}

export interface IActionRegistration {
  type: string;
  email?: string,
  password?: string,
  confirmPassword?: string,
  serverResponse?: boolean,
  errorMessage?: string,
}



