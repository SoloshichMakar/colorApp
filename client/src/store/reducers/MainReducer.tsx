import {combineReducers} from "redux";
import toDoReducer from "./toDoReducer";
import UserAuthorisationReducer from "./UserAuthorisationReducer";
import RegistrationReducer from "./RegistrationReducer";

const MainReducer = combineReducers({
    toDoReducer,
    UserAuthorisationReducer,
    RegistrationReducer
});


export default MainReducer;