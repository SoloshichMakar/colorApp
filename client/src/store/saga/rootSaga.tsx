import {all, call, fork} from 'redux-saga/effects';
import TaskContainerSaga from './TaskContainerSaga';
import UserAuthorisationSaga from './UserAuthorisationSaga';
import registrationSaga from './registrationSaga'

export default function* rootSaga () {
    yield all([
        call(TaskContainerSaga),
        call(UserAuthorisationSaga),
        call(registrationSaga)
    ]);
}