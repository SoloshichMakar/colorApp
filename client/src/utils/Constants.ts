export const COLORS: Array<string> = [
  "#f16d69",
  "#f279a2",
  "#9170cb",
  "#5eb3f6",
  "#67d7e5",
  "#ffe083",
];

export const SET_COLOR: string = "setColor";
export const ADD_TASK: string = "addTask";
export const ASYNC_ADD_TASK: string = "asyncAddTask";
export const TEXT_CHANGE: string = "textChange";
export const DELETE_TASK: string = "deleteTask";
export const ASYNC_DELETE_TASK: string = "asyncDeleteTask";

export const TASK_IS_DONE: string = "taskIsDone";
export const ASYNC_TASK_IS_DONE: string = "asyncTaskIsDone";

export const GET_DATA: string = 'getData';
export const ASYNC_GET_DATA: string = "asyncGetData";

export const TEXT_EMAIL_CHANGE: string = 'textEmailChange';
export const TEXT_PASSWORD_CHANGE: string = 'textPasswordChange';
export const USER_CHECK: string = 'userCheck';
export const ERROR: string = 'ERROR';
export const ASYNC_USER_CHECK: string = 'asyncUserCheck';
export const CREATE_USER: string = 'createUser';
export const ASYNC_CREATE_USER: string = 'asyncCreateUser';
export const TEXT_CONFIRM_PASSWORD_CHANGE: string = 'textConfirmPasswordChange';
