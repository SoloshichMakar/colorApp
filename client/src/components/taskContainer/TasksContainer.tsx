import React, { useEffect } from "react";

import {
  MainContainer,
  DeleteButton,
  Menu,
  MenuAddButton,
  MenuChooseColors,
  MenuInputTaskText,
  TaskList,
  TaskListWithDelete,
} from "./style";
import { IState, ITask } from "../../store/types/types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrash } from "@fortawesome/free-solid-svg-icons";
import { COLORS } from "../../utils/Constants";
import ChooseColorConnector from "../../store/connectors/ChooseColorConnector";
import TaskConnector from "../../store/connectors/TaskConnector";
import { Redirect } from "react-router-dom";

function getRandomInt(maxNumber: number): number {
  return Math.floor(Math.random() * Math.floor(maxNumber));
}

interface ITaskContainerProps extends IState {
  addTask: Function;
  textChange: Function;
  deleteTask: Function;
  getData: Function;
  isAuthenticated: boolean;
  userId: number;
}

const TasksContainer: React.FC<ITaskContainerProps> = ({
  tasks,
  inputText,
  chosenColor,
  addTask,
  textChange,
  deleteTask,
  getData,
  isAuthenticated,
  userId,
}) => {
  const enterKeyPress = (event: React.KeyboardEvent) => {
    if (event.key === "Enter") {
      addNewTask();
    }
  };

  const addNewTask = () => {
    if (inputText.length > 0) {
      const newTask: ITask = {
        id: "",
        taskText: inputText,
        taskColor:
          chosenColor === ""
            ? COLORS[getRandomInt(COLORS.length)]
            : chosenColor,
        completed: false,
      };
      addTask(newTask, userId);
    }
  };

  const currentTextChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const getTextAreaValue: string = e.target.value;
    textChange(getTextAreaValue);
  };

  const deleteCurrentTask = (deleteId: string) => {
    deleteTask(deleteId);
  };

  function handleGetData() {
    getData(userId);
  }

  useEffect(handleGetData, []);

  if (!isAuthenticated) {
    return <Redirect to="/login" />;
  }
  return (
    <MainContainer id="mainContainer">
      <TaskList>
        {tasks.map((task: ITask) => (
          <TaskListWithDelete>
            <TaskConnector
              key={`${task.id}`}
              taskText={task.taskText}
              taskColor={task.taskColor}
              completed={task.completed}
              id={task.id}
            />
            <DeleteButton
              className="fas fa-trash"
              onClick={() => deleteCurrentTask(task.id)}
            >
              <FontAwesomeIcon icon={faTrash} />
            </DeleteButton>
          </TaskListWithDelete>
        ))}
      </TaskList>
      <Menu>
        <MenuInputTaskText
          type="text"
          value={inputText}
          onKeyPress={enterKeyPress}
          onChange={currentTextChange}
          placeholder="Add New Item"
        />
        <MenuChooseColors>
          <ChooseColorConnector />
          <MenuAddButton className="menu__add-button" onClick={addNewTask}>
            add
          </MenuAddButton>
        </MenuChooseColors>
      </Menu>
    </MainContainer>
  );
};
export default TasksContainer;
