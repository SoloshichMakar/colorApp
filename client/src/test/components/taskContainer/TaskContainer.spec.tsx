import React from "react";
import renderer from "react-test-renderer";
import { COLORS } from "../../../utils/Constants";

import TasksContainer from "../../../components/taskContainer/TasksContainer";
import { BrowserRouter } from "react-router-dom";
import { ITask } from "../../../store/types/types";
import createSagaMiddleware from "redux-saga";
import { applyMiddleware, createStore } from "redux";
import MainReducer from "../../../store/reducers/MainReducer";
import rootSaga from "../../../store/saga/rootSaga";
import { Provider } from "react-redux";

const tasks: Array<ITask> = [
  {
    taskColor: COLORS[0],
    taskText: "Test text 1",
    id: "1",
    completed: true,
  },
  {
    taskColor: COLORS[1],
    taskText: "Test text 2",
    id: "2",
    completed: false,
  },
];

const props = {
  tasks: tasks,
  inputText: " Test text",
  chosenColor: COLORS[0],
  isAuthenticated: true,
  userId: 1,
  addTask: Function,
  textChange: Function,
  deleteTask: Function,
  getData: Function,
};

const sagaMiddleware = createSagaMiddleware();
const store = createStore(MainReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(rootSaga);

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <TasksContainer
            tasks={props.tasks}
            inputText={props.inputText}
            chosenColor={props.chosenColor}
            isAuthenticated={props.isAuthenticated}
            userId={props.userId}
            addTask={props.addTask}
            textChange={props.textChange}
            deleteTask={props.deleteTask}
            getData={props.getData}
          />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
