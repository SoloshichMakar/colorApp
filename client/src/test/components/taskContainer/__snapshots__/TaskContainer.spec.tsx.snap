// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm ePVWlZ"
  id="mainContainer"
>
  <div
    className="sc-AxgMl ccarme"
  >
    <div
      className="sc-Axmtr fttini"
    >
      <div
        className="sc-fzplWN sc-fznyAO hblrHC"
        style={
          Object {
            "backgroundColor": "lightgrey",
          }
        }
      >
        <input
          checked={true}
          className="sc-fznZeY ddeuan"
          onClick={[Function]}
          type="checkbox"
        />
        <div
          className="sc-fzplWN sc-fznKkj evBQCH"
        >
          Test text 1
        </div>
      </div>
      <a
        className="sc-AxheI hNYIgV fas fa-trash"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash fa-w-14 "
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </div>
    <div
      className="sc-Axmtr fttini"
    >
      <div
        className="sc-fzplWN sc-fznyAO hblrHC"
        style={
          Object {
            "backgroundColor": "#f279a2",
          }
        }
      >
        <input
          checked={false}
          className="sc-fznZeY ddeuan"
          onClick={[Function]}
          type="checkbox"
        />
        <div
          className="sc-fzplWN sc-fznKkj evBQCH"
        >
          Test text 2
        </div>
      </div>
      <a
        className="sc-AxheI hNYIgV fas fa-trash"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash fa-w-14 "
          data-icon="trash"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="sc-AxirZ ktywSo"
  >
    <input
      className="sc-AxiKw bqifRl"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Add New Item"
      type="text"
      value=" Test text"
    />
    <div
      className="sc-AxhCb TdRJl"
    >
      <div
        className="sc-AxmLO hjFfhP"
      >
        <label
          className="sc-fzozJi btpqE choose_colors__container"
        >
          <input
            checked={false}
            className="sc-fzpans glTXfu"
            name="choose_color"
            onClick={[Function]}
            type="radio"
            value="#f16d69"
          />
          <span
            className="sc-fzoLsD gVZEAl color_checkmark"
            style={
              Object {
                "backgroundColor": "#f16d69",
              }
            }
          />
        </label>
        <label
          className="sc-fzozJi btpqE choose_colors__container"
        >
          <input
            checked={false}
            className="sc-fzpans glTXfu"
            name="choose_color"
            onClick={[Function]}
            type="radio"
            value="#f279a2"
          />
          <span
            className="sc-fzoLsD gVZEAl color_checkmark"
            style={
              Object {
                "backgroundColor": "#f279a2",
              }
            }
          />
        </label>
        <label
          className="sc-fzozJi btpqE choose_colors__container"
        >
          <input
            checked={false}
            className="sc-fzpans glTXfu"
            name="choose_color"
            onClick={[Function]}
            type="radio"
            value="#9170cb"
          />
          <span
            className="sc-fzoLsD gVZEAl color_checkmark"
            style={
              Object {
                "backgroundColor": "#9170cb",
              }
            }
          />
        </label>
        <label
          className="sc-fzozJi btpqE choose_colors__container"
        >
          <input
            checked={false}
            className="sc-fzpans glTXfu"
            name="choose_color"
            onClick={[Function]}
            type="radio"
            value="#5eb3f6"
          />
          <span
            className="sc-fzoLsD gVZEAl color_checkmark"
            style={
              Object {
                "backgroundColor": "#5eb3f6",
              }
            }
          />
        </label>
        <label
          className="sc-fzozJi btpqE choose_colors__container"
        >
          <input
            checked={false}
            className="sc-fzpans glTXfu"
            name="choose_color"
            onClick={[Function]}
            type="radio"
            value="#67d7e5"
          />
          <span
            className="sc-fzoLsD gVZEAl color_checkmark"
            style={
              Object {
                "backgroundColor": "#67d7e5",
              }
            }
          />
        </label>
        <label
          className="sc-fzozJi btpqE choose_colors__container"
        >
          <input
            checked={false}
            className="sc-fzpans glTXfu"
            name="choose_color"
            onClick={[Function]}
            type="radio"
            value="#ffe083"
          />
          <span
            className="sc-fzoLsD gVZEAl color_checkmark"
            style={
              Object {
                "backgroundColor": "#ffe083",
              }
            }
          />
        </label>
      </div>
      <a
        className="sc-AxhUy jBdJbf menu__add-button"
        onClick={[Function]}
      >
        add
      </a>
    </div>
  </div>
</div>
`;
